agent/config/config.go

```diff
@@ -437,8 +437,26 @@ func (cfg *Config) complete() bool {
 }
 
 func fileConfig() (Config, error) {
-	fileName := utils.DefaultIfBlank(os.Getenv("ECS_AGENT_CONFIG_FILE_PATH"), defaultConfigFileName)
 	cfg := Config{}
+	fileName := os.Getenv("ECS_AGENT_CONFIG_FILE_PATH")
+	if len(fileName) == 0 {
+		fileName = defaultConfigFileName
+		// check if the default config file exists before validating it
+		_, err := os.Stat(fileName)
+		if err != nil {
+			return cfg, nil
+		}
+
+		isValidFile, err := validateConfigFile(fileName)
+		if err != nil {
+			seelog.Errorf("Unable to validate cfg file, err %v", err)
+			return cfg, err
+		}
+		if !isValidFile {
+			seelog.Error("Invalid cfg file")
+			return cfg, nil
+		}
+	}
```

Comment:
It would probably be better to have a `getConfigFileName() (string, error)` function that does this checks for Windows and for Linux just does `return utils.DefaultIfBlank(os.Getenv("ECS_AGENT_CONFIG_FILE_PATH"), defaultConfigFileName), nil`.